trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,223 @@
+    kernel_priors: bool = True,
+    trainable_likelihood: bool = False,
+    likelihood_variance: Optional[float] = None,
+) -> GPR:
+    """
+    Constructs a :class:`~gpflow.models.GPR` model with sensible initial parameters and
+    (optionally) priors. We use :class:`~gpflow.kernels.Matern52` kernel in the model as we found
+    it to be effective in most settings. We assume inputs are normalised to the unit hypercube for
+    setting sensible initial parameters for the kernel.
+
+    We set priors for kernel hyperparameters by default in order to stabilize model fitting. We
+    found the priors below to be highly effective for objective functions defined over the unit
+    hypercube. For objective functions with different scaling, other priors will likely be more
+    appropriate. Using priors allows for using maximum a posteriori estimate of these kernel
+    parameters during model fitting.
+
+    :param data: Dataset from the initial design, used for estimating the variance of observations.
+    :param trainable_likelihood: If set to `True` Gaussian likelihood paramater is set to
+        non-trainable. By default set to `False`.
+    :param priors: If set to `True` (default) priors are set for kernel parameters.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
+    else:
+        noise_variance = likelihood_variance
+
+    model = gpflow.models.GPR(data.astuple(), kernel, mean, noise_variance)
+
+    gpflow.set_trainable(model.likelihood, trainable_likelihood)
+
+    return model
+
+
+def build_sgpr(
+    data: Dataset,
+    search_space: SearchSpace,
+    kernel_priors: bool = True,
+    trainable_likelihood: bool = False,
+    likelihood_variance: Optional[float] = None,
+    num_inducing_points: Optional[int] = None,
+    trainable_inducing_points: bool = False,
+) -> SGPR:
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    inducing_points = _get_inducing_points(search_space, num_inducing_points)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
+    else:
+        noise_variance = likelihood_variance
+
+    model = gpflow.models.SGPR(
+        data.astuple(), kernel, inducing_points, mean_function=mean, noise_variance=noise_variance
+    )
+
+    gpflow.set_trainable(model.likelihood, trainable_likelihood)
+    gpflow.set_trainable(model.inducing_variable, trainable_inducing_points)
+
+    return model
+
+
+def build_vgp(
+    data: Dataset,
+    search_space: SearchSpace,
+    classification: bool = False,
```

Comment:
Maybe it makes sense to split this function? at the moment you have the ability to put `classification=True` and still define the `likelihood_variance` to something (which will be ignored).

Reply:
we opted for making it classification only, thanks!